reactive/kotlinx-coroutines-reactive/src/Await.kt

```diff
@@ -94,17 +94,19 @@ public suspend fun <T> Publisher<T>.awaitSingle(): T = awaitOne(Mode.SINGLE)
 public suspend fun <T> Publisher<T>.awaitSingleOrDefault(default: T): T = awaitOne(Mode.SINGLE_OR_DEFAULT, default)
 
 /**
- * Awaits for the single value from the given publisher or `null` value if none is emitted without blocking a thread and
- * returns the resulting value or throws the corresponding exception if this publisher had produced error.
+ * Awaits the single value from the given observable without blocking the thread and returns the resulting value, or, if
+ * this observable has produced an error, throws the corresponding exception. If more than one value or none were
+ * produced by the publisher, `null` is returned.
  *
  * This suspending function is cancellable.
- * If the [Job] of the current coroutine is cancelled or completed while this suspending function is waiting, this function
- * immediately resumes with [CancellationException].
- *
- * @throws NoSuchElementException if publisher does not emit any value
- * @throws IllegalArgumentException if publisher emits more than one value
+ * If the [Job] of the current coroutine is cancelled or completed while the suspending function is waiting, this
+ * function immediately cancels its [Subscription] and resumes with [CancellationException].
```

Comment:
Documentation and tests of `awaitSingleOrElse`and `awaitSingleOrDefault` also should be adjusted

Reply:
There's a huge commit pending with just the documentation adjustments: https://github.com/Kotlin/kotlinx.coroutines/commit/c22351c5819de286e77493f7fb5eb9cd9928f836 Would you prefer to instead deal with the documentation for all of `await*` here?

Reply:
It's much easier when the relevant changes eventually (via squash) appear as a single commit, especially when reading history/blame and investigating previous decisions.

Reply:
Sure, I just thought that it is the sweeping changes to the documentation that would be better as a single commit. Ok though, I don't mind. A question then: should the `AwaitTest` from https://github.com/Kotlin/kotlinx.coroutines/pull/2617/commits/87b6ab20171651280d09bd458ad9f2a3bd38e3a6 also be included in this PR, given that it checks the validity of the statement that is being made here?

Reply:
Yes, tests are pretty much necessary for such changes